sonar-plugin-api/src/main/java/org/sonar/api/profiles/RulesProfile.java

```diff
@@ -302,8 +303,27 @@ public ActiveRule getActiveRule(Rule rule) {
 
   /**
    * @param optionalSeverity if null, then the default rule severity is used
+   * @deprecated since 5.6 use {@link #activateRule(Rule, String)}
    */
+  @Deprecated
   public ActiveRule activateRule(final Rule rule, @Nullable RulePriority optionalSeverity) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)